test-status/api.js

```diff
@@ -0,0 +1,213 @@
+ * @return {!Object} a parameters object for github.checks.create|update.
+ */
+function createNewCheckParams(pullRequestSnapshot, type, status) {
+  const params = {
+    owner: pullRequestSnapshot.owner,
+    repo: pullRequestSnapshot.repo,
+    name: `ampproject/tests/${type}`,
+    head_sha: pullRequestSnapshot.head_sha,
+  };
+  switch (status) {
+    case 'queued':
+      Object.assign(params, {
+        status: 'queued',
+        output: {
+          title: `${type} tests are queued on Travis`,
+          summary: `The ${type} tests are queued to run on Travis. Watch ` +
+            'this space for results in a few minutes!',
+        },
+      });
+      break;
+
+    case 'started':
+      Object.assign(params, {
+        status: 'in_progress',
+        started_at: new Date().toISOString(),
+        output: {
+          title: `${type} tests are running on Travis`,
+          summary: `The ${type} tests are running on Travis. Watch this ` +
+            'space for results in a few minutes!',
+        },
+      });
+      break;
+
+    case 'skipped':
+      Object.assign(params, {
+        status: 'completed',
+        conclusion: 'neutral',
+        completed_at: new Date().toISOString(),
+        output: {
+          title: `${type} tests were not required`,
+          summary: `The ${type} tests were not required to run for this pull ` +
+            'request.',
+        },
+      });
+      break;
+
+    default:
+      throw new Error(
+          `'status' paramater is set to unexpected value ${status}`);
+  }
+  return params;
+}
+
+/**
+ * Create a parameters object for reporting the results of the tests status to
+ * an existing GitHub status check.
+ *
+ * @param {string} pullRequestSnapshot pull request snapshot from database.
+ * @param {string} type tests type slug.
+ * @param {number} checkRunId the existing check run ID.
+ * @param {number} passed number of tests that passed.
+ * @param {number} failed number of tests that failed.
+ * @return {!Object} a parameters object for github.checks.update.
+ */
+function createReportedCheckParams(
+  pullRequestSnapshot, type, checkRunId, passed, failed) {
+  const {owner, repo, head_sha: headSha} = pullRequestSnapshot;
+  const params = {
+    owner,
+    repo,
+    check_run_id: checkRunId,
+    status: 'completed',
+    completed_at: new Date().toISOString(),
+  };
+  if (failed > 0) {
+    const detailsUrl = new URL(
+        `/tests/${headSha}/${type}/status`, process.env.WEB_UI_BASE_URL);
+    Object.assign(params, {
+      details_url: detailsUrl.href,
+      conclusion: 'action_required',
+      output: {
+        title: `${failed} ${type} test${failed != 1 ? 's' : ''} failed`,
+        summary: `The ${type} tests finished running on Travis.`,
+        text: `* *${passed}* test${passed != 1 ? 's' : ''} PASSED\n` +
+          `* *${failed}* test${failed != 1 ? 's' : ''} FAILED\n\n` +
+          'Please inspect the Travis build and fix any code changes that ' +
```

Comment:
Any chance you can link to the Travis job here?

Reply:
#85